### ISSUE TITLE
Incorrect Priority Level for kube-proxy/node-problem-detector/konnectivity-server/glbc

### ISSUE THREAD

**Issue by marseel** [kubernetes/kubernetes#105370]
#### What happened:
Currently, requests from multiple components go to global-default Priority Level:
- kube-proxy
- node-problem-detector
- konnectivity-server
- glbc

#### What you expected to happen:
I expected requests assigned to workload-low or workload-high Priority Level
#### How to reproduce it (as minimally and precisely as possible):
Logs confirming issue taken from https://k8s-testgrid.appspot.com/sig-scalability-gce#gce-master-scale-performance job:
kube-proxy:
```
"HTTP" verb="GET" URI="/api/v1/nodes?fieldSelector=metadata.name%3Dgce-scale-cluster-minion-group-1-gxj1&limit=500&resourceVersion=0" latency="1.36583ms" userAgent="kube-proxy/v1.23.0 (linux/amd64) kubernetes/6f47878" audit-ID="dcde2865-8562-4d2d-9384-c1679543ebec" srcIP="10.40.8.85:54464" apf_pl="global-default" apf_fs="global-default" apf_fd="system:kube-proxy" resp=200
```
node-problem-detector:
```
I0929 17:09:18.547660      12 httplog.go:105] "HTTP" verb="GET" URI="/api/v1/nodes/gce-scale-cluster-minion-group-3-3bcw" latency="5.586004ms" userAgent="node-problem-detector/v0.8.9" audit-ID="e3950e20-5af1-4db2-a4db-4208e7a2910a" srcIP="10.40.8.153:35150" apf_pl="global-default" apf_fs="global-default" apf_fd="system:node-problem-detector" resp=200
```
konnectivity-server:
```
I0929 17:09:54.047403      12 httplog.go:105] "HTTP" verb="POST" URI="/apis/authentication.k8s.io/v1/tokenreviews" latency="1.21476ms" userAgent="proxy-server/v0.0.0 (linux/amd64) kubernetes/$Format" audit-ID="d7329045-32fc-4354-85c7-c75cb1b19ca7" srcIP="[::1]:41986" apf_pl="global-default" apf_fs="global-default" apf_fd="system:konnectivity-server" resp=201
```
glbc:
```
I0929 17:24:30.568095      12 httplog.go:105] "HTTP" verb="GET" URI="/api/v1/namespaces/test-5jtsix-3/services/medium-service-9" latency="5.498ms" userAgent="glbc/v0.0.0 (linux/amd64) kubernetes/$Format" audit-ID="00bed3ac-2efc-4506-ba4a-b29bb077a9c2" srcIP="[::1]:41830" apf_pl="global-default" apf_fs="global-default" apf_fd="system:controller:glbc" resp=200
```
#### Environment:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

**Comment by marseel** [comment]
/sig api-machinery
@kubernetes/sig-api-machinery-bugs 
/cc @MikeSpreitzer 
/cc @wojtek-t 
/cc @lavalamp

**Comment by MikeSpreitzer** [comment]
Could you run the apiserver with `-v=7` while reproducing this?  It would be helpful to see the output from the following line in `apf_filter.go`.

```
	klog.V(7).Infof("Handle(%#+v) => fsName=%q, distMethod=%#+v, plName=%q, isExempt=%v, queued=%v", requestDigest, fs.Name, fs.Spec.DistinguisherMethod, pl.Name, isExempt, queued)
```

**Comment by marseel** [comment]
Sure.

kube-proxy:
```
Handle(RequestDigest{RequestInfo: &request.RequestInfo{IsResourceRequest:true, Path:"/api/v1/services", Verb:"watch", APIPrefix:"api", APIGroup:"", APIVersion:"v1", Namespace:"", Resource:"services", Subresource:"", Name:"", Parts:[]string{"services"}}, User: &user.DefaultInfo{Name:"system:kube-proxy", UID:"uid:kube_proxy", Groups:[]string{"system:authenticated"}, Extra:map[string][]string(nil)}}) => fsName="global-default", distMethod=&v1beta2.FlowDistinguisherMethod{Type:"ByUser"}, plName="global-default", isExempt=false, queued=true
```
node-problem-detector:
```
Handle(RequestDigest{RequestInfo: &request.RequestInfo{IsResourceRequest:true, Path:"/api/v1/nodes/e2e-big-minion-group-hfrs/status", Verb:"patch", APIPrefix:"api", APIGroup:"", APIVersion:"v1", Namespace:"", Resource:"nodes", Subresource:"status", Name:"e2e-big-minion-group-hfrs", Parts:[]string{"nodes", "e2e-big-minion-group-hfrs", "status"}}, User: &user.DefaultInfo{Name:"system:node-problem-detector", UID:"uid:node-problem-detector", Groups:[]string{"system:authenticated"}, Extra:map[string][]string(nil)}}) => fsName="global-default", distMethod=&v1beta2.FlowDistinguisherMethod{Type:"ByUser"}, plName="global-default", isExempt=false, queued=true
```
konnectivity-server:
```
Handle(RequestDigest{RequestInfo: &request.RequestInfo{IsResourceRequest:true, Path:"/apis/authentication.k8s.io/v1/tokenreviews", Verb:"create", APIPrefix:"apis", APIGroup:"authentication.k8s.io", APIVersion:"v1", Namespace:"", Resource:"tokenreviews", Subresource:"", Name:"", Parts:[]string{"tokenreviews"}}, User: &user.DefaultInfo{Name:"system:konnectivity-server", UID:"uid:system:konnectivity-server", Groups:[]string{"system:authenticated"}, Extra:map[string][]string(nil)}}) => fsName="global-default", distMethod=&v1beta2.FlowDistinguisherMethod{Type:"ByUser"}, plName="global-default", isExempt=false, queued=true, Finish() => panicking=false idle=true
```
glbc:
```
Handle(RequestDigest{RequestInfo: &request.RequestInfo{IsResourceRequest:true, Path:"/api/v1/namespaces/kube-system/configmaps/ingress-uid", Verb:"get", APIPrefix:"api", APIGroup:"", APIVersion:"v1", Namespace:"kube-system", Resource:"configmaps", Subresource:"", Name:"ingress-uid", Parts:[]string{"configmaps", "ingress-uid"}}, User: &user.DefaultInfo{Name:"system:controller:glbc", UID:"uid:system:controller:glbc", Groups:[]string{"system:authenticated"}, Extra:map[string][]string(nil)}}) => fsName="global-default", distMethod=&v1beta2.FlowDistinguisherMethod{Type:"ByUser"}, plName="global-default", isExempt=false, queued=true
```

I believe this is relevant in this case: https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/gci/configure-helper.sh#L733

**Comment by MikeSpreitzer** [comment]
So for all four of those cases, the on user group from authentication was `system:authenticated`.  The username appears as the flow distinguisher.

**Comment by MikeSpreitzer** [comment]
Using `hack/local-up-cluster.sh`, I see the kube-proxy authenticated to be in the group `system:nodes`.

```
I1004 21:32:50.566372   35798 apf_filter.go:160] Handle(RequestDigest{RequestInfo: &request.RequestIn\
fo{IsResourceRequest:true, Path:"/apis/discovery.k8s.io/v1/endpointslices", Verb:"watch", APIPrefix:"apis", APIGroup:"discovery.k8s.io", APIVersion:"v1", Namespace:"", Resource:"endpointslices", Subresource:"", Name:"", Parts:[]string{"endpointslices"}}, User: &user.DefaultInfo{Name:"system:kube-proxy", UID:"", Groups:[]string{"system:nodes", "system:authenticated"}, Extra:map[string][]string(nil)}}) => fsName="system-nodes", distMethod=&v1beta2.FlowDistinguisherMethod{Type:"ByUser"}, plName="system", isExempt=false, queued=true
```

Why is the kube-proxy not in this group in the job(s) in question?

**Comment by marseel** [comment]
Our job is using `./hack/e2e-internal/e2e-up.sh`, I've also checked other job `pull-kubernetes-e2e-gce-ubuntu-containerd` (which is also using  `./hack/e2e-internal/e2e-up.sh`) and in this job `kube-proxy` is also using `global-default`.

I believe that this line is relevant when we are running e2e tests on gce: https://github.com/kubernetes/kubernetes/blob/6f942a0701bec803e68845fc1c7ceefa5dcbd8e1/cluster/gce/gci/configure-helper.sh#L756

In your script this part adds `system:nodes` to kube-proxy: https://github.com/kubernetes/kubernetes/blob/6f942a0701bec803e68845fc1c7ceefa5dcbd8e1/hack/local-up-cluster.sh#L487

So definitely there is discrepancy. 
I'm just wondering if we want to depend on kube-proxy and other components having this group or maybe we want to add FlowSchema to make sure it's in correct PL. I'm not sure how common this problem is in other setups though.

**Comment by marseel** [comment]
/cc @bowei 
/cc @robscott 
Tldr: In current GCE setup of k8s kube-proxy user is not in "system:nodes" group. This affects how kube-proxy requests are assigned to Priority Levels In P&F. In terms of kubernetes, do we want to make requirement that kube-proxy user is in "system:nodes" group and fix current GCE setup or maybe add separate FlowSchema for kube-proxy that doesn't depend on user group.  WDYT?

**Comment by MikeSpreitzer** [comment]
First let's be clear about responsibilities.
- kube-proxy is in k/k and should be supported in k/k.
- node-problem-detector and konnectivity-server are _not_ in k/k but _are_ elsewhere in github.com/kubernetes and should be supported by k/k.
- glbc is _not_ anywhere under github.com/kubernetes and should be supported by whoever owns/deploys glbc.

**Comment by MikeSpreitzer** [comment]
I can sympathize with the argument that Kubernetes should support very granular authentication and authorization, distinguishing between kubelet and kube-proxy in general and even distinguishing individuals (i.e., each kubelet and each kube-proxy runs with a different username).  If each kube-proxy authenticates with a distinct username then we are going to need a common user-group for all the kube-proxys.  Same for node-problem-detector and konnectivity-server.

**Comment by lavalamp** [comment]
kube-proxy isn't really a node, so it doesn't seem quite right for it to be in the system:nodes group.

kube-proxy and npd both run *on* nodes, konnectivity-server runs with the control plane.

I think a group for these things makes more sense than managing an FSC for each one. Maybe SIG Auth can say if making such a thing is hard or not?

/sig auth

**Comment by MikeSpreitzer** [comment]
There is an additional reason for each kube-proxy, node-problem-detector, and konnectivity-server to authenticate with a distinct username: so that they can be classified to different flows.

**Comment by MikeSpreitzer** [comment]
Oh, I was confused.  Yes, the konnectivity-servers run on control plane nodes.  I say plural because I think we should be prepared for multiple of them acting in parallel.  So the konnectivity-server still has the potential issue of authenticating individuals.

**Comment by jpbetz** [comment]
@MikeSpreitzer should we keep this open?

**Comment by MikeSpreitzer** [comment]
@jpbetz: Yes, I think so.

**Comment by aojea** [comment]
> glbc is not anywhere under github.com/kubernetes and should be supported by whoever owns/deploys glbc.

is in kubernetes/ingress-gce :)
